src/core/meshcop/tcat_agent.cpp

```diff
@@ -540,6 +547,95 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     return error;
 }
 
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage)
+{
+    Error       error  = kErrorNone;
+    const char *name   = otThreadGetNetworkName(&GetInstance());
+    uint16_t    length = static_cast<uint16_t>(strlen(name));
+    ot::Tlv     tlv;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+    VerifyOrExit(length > 0, error = kErrorInvalidState);
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(name, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage)
+{
+    uint8_t deviceId[8];
```

Comment:
Note: the length needs to be defined by a compile-time flag! See the spec: it should be easy to set/update by the vendor to any desired length

Reply:
@EskoDijk what is the correlation of this vendor ID and the one we advertise?
Since the max value of the vendor ID there is 5 and I reused those here.
There is no problem in creating new API to set this value separately to a different value than the advertised ones. However this makes sense if we decide that this vendor ID and advertised ones aren't related at all.

Reply:
@canisLupus1313 the Device Discriminator is indeed up to 5 B - this is a different number than the unique Device ID. The Discriminator can be the hash of the ID for example, and is not necessarily unique - it hides the true identity in any case. (The rest of the advertised IDs are the Vendor OUI or PEN which are all different.)